.github/workflows/_shared-first-time-setup.yml

```diff
@@ -0,0 +1,50 @@
+name: "Repository First Time Setup automation (shared)"
+
+on:
+  workflow_call:
+    inputs:
+        esf_version:
+          type: string
+          default: 'https://changeme'
+          required: true
+        groupId:
+          type: string
+          default: ''
+          required: true
+        artifactId:
+          type: string
+          default: ''
+          required: true
+
+jobs:
+  init:
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Download ESF archetype
+      uses: carlosperate/download-file-action@v1
+      with:
+        file-url: ${{ github.event.inputs.esf_version }}
+      if: startsWith(${{ github.event.inputs.esf_version }}, 'https://eth-repo.s3.amazonaws.com/esf')
+
+    - name: Run the maven install
+      run: |
+        mvn install:install-file -Dfile=esf-addon-archetype-${{ github.event.inputs.esf_version }}.jar \
+        -DgroupId=com.eurotech.framework -DartifactId=esf-addon-archetype \
+        -Dversion=${{ github.event.inputs.esf_version }} -Dpackaging=jar
```

Comment:
Since `github.event.inputs.esf_version` is a link `-Dversion=${{ github.event.inputs.esf_version }}` is not going to work as expected

Reply:
... neither `file=esf-addon-archetype-${{ github.event.inputs.esf_version }}.jar`

Reply:
We might be able to parse the link and grab the ESF version from that one, what do you think?